### PR TITLE
Some general cleanup

### PR DESCRIPTION
**Pull request by harrywhite4**
The major functional changes here are:
- Using a standard logger name
- Simplifying the context manager to use reset and only set language on enter

### CODE REVIEW

File: garnett/context.py [hunk 1]
```diff
@@ -1,23 +1,17 @@
...
     def __enter__(self):
-        pass
+        self.token = _ctx_language.set(self.language)
 
     def __exit__(self, exc_type, exc_value, traceback):
-        # Using Context.reset like shown below causes weird errors
-        # TODO: Maybe fix this?
-        # ctx_language.reset(self.language)
-        ctx_language.set(self.old_language)
+        _ctx_language.reset(self.token)
```
Comment (LegoStormtroopr) [inline garnett/context.py:17]: If this works - awesome, but I had difficulties getting it to work

File: garnett/context.py [hunk 2]
```diff
@@ -1,23 +1,17 @@
...
-
-ctx_language = contextvars.ContextVar("garnett_language")
+# Internal context var should be set via set_field_language and get via get_current_language
+_ctx_language = contextvars.ContextVar("garnett_language")
 
 
 class set_field_language(ContextDecorator):
-    def __init__(self, language, deactivate=False):
-        self.old_language = ctx_language.set(language).old_value
-        if self.old_language == contextvars.Token.MISSING:
```
Comment (LegoStormtroopr) [inline garnett/context.py:12]: What happens if there was no previous language?

**Review by LegoStormtroopr — APPROVED** [review]

File: garnett/context.py (hunk 1 above)
Comment (harrywhite4) [reply garnett/context.py:17]: I have tested this and it's working nicely. Problem was you need to pass the output of set directly to reset

File: garnett/context.py (hunk 2 above)
Comment (harrywhite4) [reply garnett/context.py:12]: This is handled by the `get_current_language` function which just returns the default language

**Pull request merged**